src/main/kotlin/org/jetbrains/research/libsl/type/TypeInferrer.kt

```diff
@@ -46,6 +50,18 @@ class TypeInferer(private val context: LslContextBase) {
         }
     }
 
+    private fun getArrayLiteralType(arrayLiteral: ArrayLiteral): Type {
+        val typeOfElements = arrayLiteral.value.fold(anyType) { acc, expression ->
```

Comment:
I don't think it is correct to start resolving the type of an expression straight from `Any` type.
There probably should be some `None` or `Empty` type, from which the process starts, but should never end (by throwing an exception). _Note:_ **no variable or constant** should end up having such type.

At the same time I imagine a case when there is a variable declaration without a type specified, which also equals to an empty array, like this: `var x = [];` In such scenario the variable should end up having type `array<any>`.